panic: runtime error: invalid memory address or nil pointer dereference

When i try to run code with **goccy/go-json**, i get this error:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x4f5d4d]

goroutine 1 [running]:
github.com/goccy/go-json/internal/encoder/vm.ptrToString(...)
        /home/efectn/go/pkg/mod/github.com/goccy/go-json@v0.8.0/internal/encoder/vm/util.go:89
github.com/goccy/go-json/internal/encoder/vm.Run(0xc0000a2680, {0xc0000e6000, 0x0, 0x400}, 0xc0000ba5c0)
        /home/efectn/go/pkg/mod/github.com/goccy/go-json@v0.8.0/internal/encoder/vm/vm.go:3679 +0x2438d
github.com/goccy/go-json.encodeRunCode(0xc00009ae00, {0xc0000e6000, 0x639028, 0xc0000ba040}, 0xc0000e6400)
        /home/efectn/go/pkg/mod/github.com/goccy/go-json@v0.8.0/encode.go:307 +0x68
github.com/goccy/go-json.encode(0xc0000a2680, {0x505460, 0xc0000be050})
        /home/efectn/go/pkg/mod/github.com/goccy/go-json@v0.8.0/encode.go:232 +0x21c
github.com/goccy/go-json.marshal({0x505460, 0xc0000be050}, {0x0, 0x0, 0x7f3de50c1701})
        /home/efectn/go/pkg/mod/github.com/goccy/go-json@v0.8.0/encode.go:147 +0xba
github.com/goccy/go-json.MarshalWithOption(...)
        /home/efectn/go/pkg/mod/github.com/goccy/go-json@v0.8.0/json.go:186
github.com/goccy/go-json.Marshal({0x505460, 0xc0000be050})
        /home/efectn/go/pkg/mod/github.com/goccy/go-json@v0.8.0/json.go:171 +0x2a
main.main()
        /home/efectn/go/src/goccy_test/main.go:40 +0x45d
exit status 2
```

But i can get results with **encoding/json**:
`2021/12/04 21:14:15 {"account":"saltfish","password":"123456","nickname":"user1","hubs":[{"hid":"alpha","name":"hub","size":100}],"friends":[{"account":"saltfish","password":"123456","nickname":"user2"}]}`


Code: 
```
package main

import (
	"github.com/goccy/go-json"
	"log"
)

type User struct {
	Account  *string `json:"account" validate:"required"`
	Password *string `json:"password" validate:"required"`
	Nickname *string `json:"nickname" validate:"required"`
	Address  *string `json:"address,omitempty"`

	Hubs    []*Hub  `json:"hubs,omitempty"`
	Friends []*User `json:"friends,omitempty"`
}

type Hub struct {
	HID  *string `json:"hid" validate:"required"`
	Name *string `json:"name" validate:"required"`
	Size *int    `json:"size" validate:"required"`
}

func main() {
	user1Account, user1Password, user1Nickname := "saltfish", "123456", "user1"
	user1 := &User{Account: &user1Account, Password: &user1Password, Nickname: &user1Nickname, Address: nil}

	user2Account, user2Password, user2Nickname := "saltfish", "123456", "user2"
	user2 := &User{Account: &user2Account, Password: &user2Password, Nickname: &user2Nickname, Address: nil}

	hubID, hubName, hubSize := "alpha", "hub", 100
	hub := &Hub{HID: &hubID, Name: &hubName, Size: &hubSize}
	hubs := []*Hub{hub}
	friends := []*User{user2}

	user1.Hubs = hubs
	user1.Friends = friends

	data, _ := json.Marshal(user1)
	log.Println(string(data))
}

```

Relates Issue: https://github.com/gofiber/fiber/issues/1628

--- a/encode_test.go
+++ b/encode_test.go
@@ -2242,3 +2242,39 @@ func TestIssue299(t *testing.T) {
 		}
 	})
 }
+
+func TestRecursivePtrHead(t *testing.T) {
+	type User struct {
+		Account  *string `json:"account"`
+		Password *string `json:"password"`
+		Nickname *string `json:"nickname"`
+		Address  *string `json:"address,omitempty"`
+		Friends  []*User `json:"friends,omitempty"`
+	}
+	user1Account, user1Password, user1Nickname := "abcdef", "123456", "user1"
+	user1 := &User{
+		Account:  &user1Account,
+		Password: &user1Password,
+		Nickname: &user1Nickname,
+		Address:  nil,
+	}
+	user2Account, user2Password, user2Nickname := "ghijkl", "123456", "user2"
+	user2 := &User{
+		Account:  &user2Account,
+		Password: &user2Password,
+		Nickname: &user2Nickname,
+		Address:  nil,
+	}
+	user1.Friends = []*User{user2}
+	expected, err := stdjson.Marshal(user1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	got, err := json.Marshal(user1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !bytes.Equal(expected, got) {
+		t.Fatalf("failed to encode. expected %q but got %q", expected, got)
+	}
+}

--- a/internal/cmd/generator/main.go
+++ b/internal/cmd/generator/main.go
@@ -120,7 +120,7 @@ func (t OpType) HeadToOmitEmptyHead() OpType {
 }
 
 func (t OpType) PtrHeadToHead() OpType {
-  idx := strings.Index(t.String(), "Ptr")
+  idx := strings.Index(t.String(), "PtrHead")
   if idx == -1 {
     return t
   }

--- a/internal/encoder/optype.go
+++ b/internal/encoder/optype.go
@@ -894,7 +894,7 @@ func (t OpType) HeadToOmitEmptyHead() OpType {
 }
 
 func (t OpType) PtrHeadToHead() OpType {
-	idx := strings.Index(t.String(), "Ptr")
+	idx := strings.Index(t.String(), "PtrHead")
 	if idx == -1 {
 		return t
 	}

